src/app/common/transactions/supported-transactions.ts

```diff
@@ -0,0 +1,28 @@
+import { BitcoinTransaction } from './bitcoin/transaction.models';
+import { StacksTransactions } from './stacks/transaction.models';
+
+export enum SupportedBlockchainTransactionTypes {
+  bitcoin = 'bitcoin',
+  stacks = 'stacks',
+}
+
+export type SupportedTransactionTypes = keyof typeof SupportedBlockchainTransactionTypes;
+export type SupportedTransactions = StacksTransactions | BitcoinTransaction;
+
+function isBitcoinTransaction(txType: SupportedTransactionTypes) {
+  return txType === SupportedBlockchainTransactionTypes.bitcoin;
+}
+
+function isStacksTransaction(txType: SupportedTransactionTypes) {
+  return txType === SupportedBlockchainTransactionTypes.stacks;
+}
+
+type SupportedTransactionTypeMap<T> = Record<SupportedTransactionTypes, T>;
+
+export function whenSupportedTransaction(txType: SupportedTransactionTypes) {
```

Comment:
I don't understand this? We don't have the chain Id on the transaction. How is this different than `whenWallet` or `whenPageMode`? Maybe could be `whenSupportedTransactionType`?

Reply:
~~Thinking out loud, but another way to look at it might be to have supported chains, rather than supported tx types. Could these types all be SupportedChain named?~~

Yeah sorry forget that, doesn't make sense (there's not stx or bitcoin mode). 

How about `whenTransactionFromChain({ bitcoin: '', stacks: '' })`. I guess my dislike of using "supported" is that there are no "unsupported transactions". Isn't a transaction being in the wallet in the first place implicit that it's supported.

That's to say, `SupportedBlockchainTransactionTypes` makes sense, but I reckon we don't need to continue to use the term beyond it.

Reply:
Ok, got it, makes sense. I'll change.